routes/people-routes.js

```diff
@@ -0,0 +1,77 @@
+// *********************************************************************************
+// api-routes.js - this file offers a set of routes for displaying and saving data to the db
+// *********************************************************************************
+
+// Dependencies
+// =============================================================
+
+// Requiring our models
+var db = require("../models");
+
+// Routes
+// =============================================================
+module.exports = function(app) {
+
+  // GET route for getting all of the people
+  app.get("/api/people", function(req, res) {
+    var query = {};
+    if (req.query.location_id) {
+      query.LocationId = req.query.location_id;
+    }
+    // Here we add an "include" property to our options in our findAll query
+    // We set the value to an array of the models we want to include in a left outer join
+    // In this case, just db.Location
+    db.Post.findAll({
+      where: query,
+      include: [db.Location]
+    }).then(function(dbPeople) {
+      res.json(dbPeople);
+    });
+  });
+
+  // Get route for retrieving a single post
+  app.get("/api/people/:id", function(req, res) {
+    // Here we add an "include" property to our options in our findOne query
+    // We set the value to an array of the models we want to include in a left outer join
+    // In this case, just db.Location
+    db.Post.findOne({
+      where: {
+        id: req.params.id
+      },
+      include: [db.Location]
+    }).then(function(dbPeople) {
+      res.json(dbPeople);
+    });
+  });
+
+  // POST route for saving a new post
+  app.post("/api/people", function(req, res) {
+    db.Post.create(req.body).then(function(dbPeople) {
+      res.json(dbPeople);
+    });
+  });
+
+  // DELETE route for deleting people
+  app.delete("/api/people/:id", function(req, res) {
+    db.Post.destroy({
+      where: {
+        id: req.params.id
+      }
+    }).then(function(dbPeople) {
+      res.json(dbPeople);
+    });
+  });
+
+  // PUT route for updating people
+  app.put("/api/people", function(req, res) {
+    db.Post.update(
+      req.body,
```

Comment:
Don't send req.body incase the API changes in the future, pull off the properties you need.

Reply:
for example:
const person = {
  name: req.body.name,
  hobbies: req.body.hobbies
};
db.Post.update(person,...